net/client.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * General-purpose TCP client, to be used with corresponding Net::Server object.
+ * Various sink/source-type objects are used for either sending or receiving
+ * data from the server.
+ */
+
+#pragma once
+
+// C++ includes
+#include <limits>
+#include <string>
+#include <vector>
+
+// OpenCV
+#include <opencv2/opencv.hpp>
+
+// local includes
+#include "node.h"
+#include "socket.h"
+
+namespace GeNNRobotics {
+namespace Net {
+class Client
+  : public Node
+  , Socket
+{
+public:
+    Client(const std::string host, int port = DefaultListenPort);
```

Comment:
I feel this should be a reference rather than a copy?

Reply:
Good spot!